### PR TITLE
[Feature] Add api key and ssl to http server

### PR DESCRIPTION
**Pull request by AllentDan**

### CODE REVIEW

**Review by hanckjt — APPROVED** [review]

**Comment by lvhan028** [comment]
Please update user guide, introducing this feature

File: lmdeploy/serve/openai/api_client.py [hunk 1]
```diff
@@ -35,16 +35,25 @@ class APIClient:
     Args:
         api_server_url (str): communicating address 'http://<ip>:<port>' of
             api_server
+        api_key (str | None): api key. Default to None, which means no
+            api key will be used.
     """
 
-    def __init__(self, api_server_url: str, **kwargs):
+    def __init__(self,
```
[the remark this answers is not in the document]
Comment (AllentDan) [reply lmdeploy/serve/openai/api_client.py:42]: Updated.

File: lmdeploy/cli/utils.py [hunk 2]
```diff
@@ -229,6 +229,26 @@ def log_level(parser):
                                    choices=list(logging._nameToLevel.keys()),
                                    help='Set the log level')
 
+    @staticmethod
+    def api_keys(parser):
+        return parser.add_argument(
+            '--api-keys',
```
Comment (RunningLeon) [inline lmdeploy/cli/utils.py:235]: suggest use `nargs='*'` for multiple inputs.

File: lmdeploy/cli/serve.py [hunk 3]
```diff
@@ -158,6 +160,10 @@ def add_parser_api_client():
         parser.add_argument('api_server_url',
                             type=str,
                             help='The URL of api server')
+        parser.add_argument('--api-key',
```
Comment (RunningLeon) [inline lmdeploy/cli/serve.py:163]: could make --api-key and --api-keys into one for serve and client

**Review by lvhan028 — APPROVED** [review]

**Review by RunningLeon — APPROVED** [review]
LGTM

**Comment by seetimee** [comment]
if min_latency < latency:这个判断条件是不是有点问题,应该是反过来吧

**Pull request merged**